### PR TITLE
Fix conflicts and bugs

### LINKED ISSUE #5: TypeError: Shapes must be 1D sequences of concrete values of integer type
image

Running with #4

### PR DESCRIPTION
**Pull request by junhoyeo**
Closes #5 

- This PR fix the conflicts between master and #6 - You can merge this PR first and the commits in #6 will also be automatically rebased, closing both PRs.

- Some typos and bugs are fixed, refactors have been made.

### CODE REVIEW

File: rA9/optim/adam.py [hunk 1]
```diff
@@ -21,7 +22,7 @@ def step(self, closure=None):
                     state['step'] = 0
                     state['exp_avg'] = jnp.zeros(p.shape)
                     state['exp_avg_sq'] = jnp.zeros(p.shape)
-                d_w= d_weights[count]
+                d_w = self.d_weights[count]
```
Comment (junhoyeo) [inline rA9/optim/adam.py:25]: The property `d_weights` have been used in `Adam.step` even though it was not preserved, so I stored it in `self`.

File: rA9/synapses/Conv.py [hunk 2]
```diff
@@ -93,33 +101,32 @@ def jnp_fn(input_jnp, weights_jnp, bias=None, stride=1, padding=0):
...
+            X_col = im2col_indices(input_jnp, h_filter,
+                                   w_filter, padding=padding, stride=stride)
             W_col = weights_jnp.reshape(n_filters, -1)
 
             out = jnp.matmul(W_col, X_col)
 
-            if b is not None:
-                out += b
+            if bias is not None:
+                out += bias
```
Comment (junhoyeo) [inline rA9/synapses/Conv.py:111]: You made a typo, `bias` to `b`, so it was renamed.

File: rA9/synapses/Linear.py [hunk 3]
```diff
@@ -40,24 +46,32 @@ def __init__(self, in_features, out_features):
     def reset_parameters(self):
         size = self.weight.data.shape
         stdv = 1. / math.sqrt(size[1])
-        self.weight = jax.random.uniform(minval=-stdv, maxval=stdv, shape=self.weight, key=keyW)
+
+        keyW = jax.random.PRNGKey(0)
+        keyB = jax.random.PRNGKey(0)
```
Comment (junhoyeo) [inline rA9/synapses/Linear.py:51]: The both keys used in `jax.random.uniform` was omitted, so it was added.

File: rA9/synapses/Conv.py [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
-from ..networks.module import Module
-from jax import random, vjp, jit, linear_util as lu
 from functools import wraps
 
-from jax.api import argnums_partial, _check_scalar
+import jax
+import jax.numpy as jnp
+from jax import jit
+from jax import linear_util as lu
+from jax import random, value_and_grad, vjp
```
Comment (junhoyeo) [inline rA9/synapses/Conv.py:8]: The import for `value_and_grad` was missing in some files, so it was added.
I also sorted imports for better readability for developers!

File: rA9/synapses/Conv.py (hunk 2 above)
Comment (Dongyeongkim) [reply rA9/synapses/Conv.py:111]: This project doesn't use bias so it will be eradicated

**Review by Dongyeongkim — APPROVED** [review]
Need to change some sort of codes but looking really well in this time!

File: rA9/synapses/Linear.py (hunk 3 above)
Comment (Dongyeongkim) [reply rA9/synapses/Linear.py:51]: We don't need bias key so it will be eradicated(KeyB)

File: rA9/synapses/Conv.py (hunk 2 above)
Comment (junhoyeo) [reply rA9/synapses/Conv.py:111]: I was going to say you should also note Linear.py too, but you caught me under this comment lol

Comment (Dongyeongkim) [reply rA9/synapses/Conv.py:111]: LOL

**Pull request merged**